### PR TITLE
MM-34910 New messages toast should only count root posts 

### PR DESCRIPTION
**Pull request by reflog**
#### Summary

unread badge in channel should count root posts only

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-34910

#### Related Pull Requests

- Has server changes (please link here)
- Has redux changes (please link here)
- Has mobile changes (please link here)

#### Screenshots

```release-note
NONE
```

### CODE REVIEW

File: components/toast_wrapper/index.jsx [hunk 1]
```diff
@@ -10,19 +10,40 @@ import {Posts} from 'mattermost-redux/constants';
...
 
 export function makeCountUnreadsBelow() {
     return createSelector(
         getAllPosts,
         getCurrentUserId,
+        isCollapsedThreadsEnabled,
         (state, postIds) => postIds,
         (state, postIds, lastViewedBottom) => lastViewedBottom,
-        (allPosts, currentUserId, postIds, lastViewedBottom) => {
+        (allPosts, currentUserId, postIds, lastViewedBottom, isCollapsed) => {
```
Comment (ashishbhate) [inline components/toast_wrapper/index.jsx:46]: The arguments to the final result function are out of order

Comment (reflog) [reply components/toast_wrapper/index.jsx:46]: damn! nice catch @ashishbhate

**Review by ashishbhate — APPROVED** [review]

**Review by ogi-m — APPROVED** [review]
LGTM 👍🏻

File: components/toast_wrapper/index.jsx [hunk 2]
```diff
@@ -10,19 +10,40 @@ import {Posts} from 'mattermost-redux/constants';
...
+        getAllPosts,
+        getCurrentUserId,
+        getCurrentChannel,
+        (allPosts, currentUserId, channel) => {
+            // Count the number of new posts made by other users that haven't been deleted and are root posts
+            return Object.values(allPosts).filter((post) => {
+                return post.root_id === '' &&
+                    post.channel_id === channel.id &&
+                    post.user_id !== currentUserId &&
+                    post.state !== Posts.POST_DELETED;
```
Comment (calebroseland) [inline components/toast_wrapper/index.jsx:30]: With the ability to arbitrarily mark a post/channel as unread (including before your own root posts), I think currentUser posts should be included in the count.

```suggestion
                return (
                    post.root_id === '' &&
                    post.channel_id === channel.id &&
                    post.state !== Posts.POST_DELETED
                );
```

File: components/toast_wrapper/index.jsx [hunk 3]
```diff
@@ -10,19 +10,40 @@ import {Posts} from 'mattermost-redux/constants';
...
 import {updateToastStatus} from 'actions/views/channel';
 
 import ToastWrapper from './toast_wrapper.jsx';
+export function makeGetRootPosts() {
+    return createSelector(
+        getAllPosts,
+        getCurrentUserId,
+        getCurrentChannel,
+        (allPosts, currentUserId, channel) => {
+            // Count the number of new posts made by other users that haven't been deleted and are root posts
```
Comment (calebroseland) [inline components/toast_wrapper/index.jsx:25]:
```suggestion
            // Count the number of new posts that haven't been deleted and are root posts
```

**Review by calebroseland — CHANGES_REQUESTED** [review]

**Review by calebroseland — APPROVED** [review]

**Pull request merged**